client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content/hooks/use-wp-plan-name.ts

```diff
@@ -4,26 +4,16 @@ import {
 	PLAN_PREMIUM,
 	PLAN_PERSONAL,
 } from '@automattic/calypso-products';
-import { useTranslate } from 'i18n-calypso';
+import { getPlan } from '@automattic/calypso-products/src';
 
-const useWPPlanName = ( plans: Array< string > ) => {
-	const translate = useTranslate();
+const PLANS = [ PLAN_BUSINESS, PLAN_ECOMMERCE, PLAN_PREMIUM, PLAN_PERSONAL ];
 
+const useWPPlanName = ( plans: Array< string > ) => {
 	const getPlanName = () => {
 		for ( const plan of plans ) {
-			switch ( true ) {
-				// Plans could be either monthly, 1 year, 2 years, or 3 years.
-				// So we need to check if the plan starts with the plan slug.
-				case plan.startsWith( PLAN_BUSINESS ):
-					return translate( 'WordPress.com Business Plan' );
-				case plan.startsWith( PLAN_ECOMMERCE ):
-					return translate( 'WordPress.com Commerce Plan' );
-				case plan.startsWith( PLAN_PREMIUM ):
-					return translate( 'WordPress.com Premium Plan' );
-				case plan.startsWith( PLAN_PERSONAL ):
-					return translate( 'WordPress.com Personal Plan' );
-				default:
-					return '';
+			const matchingPlan = PLANS.find( ( planConstant ) => plan.startsWith( planConstant ) );
+			if ( matchingPlan ) {
+				return getPlan( matchingPlan )?.getTitle();
```

Comment:
Instead of using just the product name, we can maybe use the translation string `WordPress.com %(name)s Plan` to make it clearer that we are referring to a plan type. wdyt?

Reply:
Yes, we can. I'm just waiting for Simon's suggestions here.